### PR TITLE
install: Some cleanups around root_path

### PR DESCRIPTION
**Pull request by cgwalters**
The extra cloning we were doing here looked odd to me. I think this ends up being cleaner make the toplevel variable mutable and replace it only in the circumstance we detect an ostree deployment.

### CODE REVIEW

File: lib/src/install.rs [hunk 1]
```diff
@@ -1636,40 +1636,42 @@ pub(crate) async fn install_to_filesystem(
...
 
+    // If we're installing to an ostree root, then find the physical root from
+    // the deployment root.
     let possible_physical_root = fsopts.root_path.join("sysroot");
     let possible_ostree_dir = possible_physical_root.join("ostree");
-    let root_path = if possible_ostree_dir.exists() {
+    if possible_ostree_dir.exists() {
         tracing::debug!(
-            "ostree detected in {possible_ostree_dir}, assuming / is a deployment root and using {possible_physical_root} instead of {root_path} as target root"
+            "ostree detected in {possible_ostree_dir}, assuming / is a deployment root and using {possible_physical_root} as target root"
```
Comment (omertuc) [inline lib/src/install.rs:1658]: Maybe even

```suggestion
            "ostree detected in {possible_ostree_dir}, assuming {root_path} is a deployment root and using {possible_physical_root} as target root"
```

**Review by omertuc — CHANGES_REQUESTED** [review]
Looks good

**Comment by omertuc** [comment]
>The extra cloning we were doing here looked odd to me. I think this ends up being cleaner make the toplevel variable mutable and replace it only in the circumstance we detect an ostree deployment.

I personally find mutations harder to follow as a code reader/debugger, even if they're sometimes "simpler" (mostly in terms of LoC) and more performant, but it's all very subjective

**Review by omertuc — APPROVED** [review]

File: lib/src/install.rs [hunk 2]
```diff
@@ -1636,40 +1636,42 @@ pub(crate) async fn install_to_filesystem(
...
 
+    // If we're installing to an ostree root, then find the physical root from
+    // the deployment root.
     let possible_physical_root = fsopts.root_path.join("sysroot");
     let possible_ostree_dir = possible_physical_root.join("ostree");
-    let root_path = if possible_ostree_dir.exists() {
+    if possible_ostree_dir.exists() {
         tracing::debug!(
-            "ostree detected in {possible_ostree_dir}, assuming / is a deployment root and using {possible_physical_root} instead of {root_path} as target root"
+            "ostree detected in {possible_ostree_dir}, assuming {root_path} is a deployment root and using {possible_physical_root} as target root"
```
Comment (omertuc) [inline lib/src/install.rs:1658]: Oh sorry, there's no `root_path`

**Review by omertuc — CHANGES_REQUESTED** [review]

**Comment by cgwalters** [comment]
> I personally find mutations harder to follow as a code reader/debugger, even if they're sometimes "simpler" (mostly in terms of LoC) and more performant, but it's all very subjective

Agreed that the mutation can be confusing, but I think in this case it is clearer than creating a shadowed copy of the same value.

However I have some other followups here

**Pull request merged**